### PR TITLE
[receiver/snmpreceiver] Add boilerplate skeleton for SNMP metric receiver

### PR DESCRIPTION
**Pull request by StefanKurek**
**Description:**
This is the boilerplate setup of the SNMP Receiver. It collects without erroring, but does not have any useful config or actually collect any SNMP data at this point.

**Link to tracking Issue:**
[13409](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/13409)

**Testing:**
No unit or integration tests. Just manually ran to see if it worked.

**Documentation:**
README added for what the SNMP receiver will ultimately be.

### CODE REVIEW

File: receiver/snmpreceiver/README.md [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        column_oids:
+          - oid: "1.1.1.2"
+            resource_attributes:
+              - resource_attr.name.2
+
+```
+
+The full list of settings exposed for this receiver are documented [here](./config.go) with detailed sample configurations [here](./testdata).
+
+[beta]:https://github.com/open-telemetry/opentelemetry-collector#in-development
```
Comment (djaglowski) [inline receiver/snmpreceiver/README.md:244]:
```suggestion
[in development]:https://github.com/open-telemetry/opentelemetry-collector#in-development
```

File: receiver/snmpreceiver/README.md [hunk 2]
```diff
@@ -0,0 +1,245 @@
+# SNMP Receiver
+
+| Status                   |           |
+| ------------------------ |-----------|
+| Stability                | [in-development] |
```
Comment (djaglowski) [inline receiver/snmpreceiver/README.md:5]:
```suggestion
| Stability                | [in development] |
```

**Review by djaglowski — APPROVED** [review]
Looks good. Just two doc nits

**Comment by djaglowski** [comment]
Looks like it needs a `make gotidy` as well

File: receiver/snmpreceiver/README.md [hunk 3]
```diff
@@ -0,0 +1,245 @@
...
+      # Each resource will have a resource attribute with a hopefully unique value.
+      metric.name.5:
+        unit: "By"
+        gauge:
+          value_type: int
+        column_oids:
+          - oid: "1.1.1.2"
+            resource_attributes:
+              - resource_attr.name.2
+
```
[the remark this answers is not in the document]
Comment (dmitryax) [reply receiver/snmpreceiver/README.md:239]: NVM. I confused it with `metadata.yaml`

**Comment by djaglowski** [comment]
It appears that `internal/components/TestDefaultReceivers` is failing:
```
=== RUN   TestDefaultReceivers
    receivers_test.go:358: 
        	Error Trace:	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/components/receivers_test.go:358
        	Error:      	"[{active_directory_ds %!s(bool=true) %!s(components.getReceiverConfigFn=<nil>)} {aerospike %!s(bool=false) %!s(components.getReceiverConfigFn=<nil>)} {apache %!s(bool=false) %!s(components.getReceiverConfigFn=<nil>)} ...]" should have 73 item(s), but has 72
        	Test:       	TestDefaultReceivers
        	Messages:   	All receivers must be added to the lifecycle suite
```

**Comment by djaglowski** [comment]
Link to failure: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3191785503/jobs/5208530885#step:8:9653

**Comment by StefanKurek** [comment]
> Link to failure: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/3191785503/jobs/5208530885#step:8:9653

@djaglowski ok, so the list of tests was always cut off for me. i had to expand/collapse them a couple times to get everything to show up.

**Pull request merged**